<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 2 GWAS 2 - Association analysis | Algorithms in Bioinformatics and Genomics</title>
  <meta name="description" content="Chapter 2 GWAS 2 - Association analysis | Algorithms in Bioinformatics and Genomics" />
  <meta name="generator" content="bookdown 0.42.1 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 2 GWAS 2 - Association analysis | Algorithms in Bioinformatics and Genomics" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 2 GWAS 2 - Association analysis | Algorithms in Bioinformatics and Genomics" />
  
  
  

<meta name="author" content="Clara Tang" />


<meta name="date" content="2026-02-11" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="gwas-1---data-management-and-quality-control.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">HTI5802 Practical</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome</a></li>
<li class="chapter" data-level="1" data-path="gwas-1---data-management-and-quality-control.html"><a href="gwas-1---data-management-and-quality-control.html"><i class="fa fa-check"></i><b>1</b> GWAS 1 - Data management and Quality control</a>
<ul>
<li class="chapter" data-level="1.1" data-path="gwas-1---data-management-and-quality-control.html"><a href="gwas-1---data-management-and-quality-control.html#objectives"><i class="fa fa-check"></i><b>1.1</b> Objectives</a></li>
<li class="chapter" data-level="1.2" data-path="gwas-1---data-management-and-quality-control.html"><a href="gwas-1---data-management-and-quality-control.html#prerequisites"><i class="fa fa-check"></i><b>1.2</b> Prerequisites</a></li>
<li class="chapter" data-level="1.3" data-path="gwas-1---data-management-and-quality-control.html"><a href="gwas-1---data-management-and-quality-control.html#dataset"><i class="fa fa-check"></i><b>1.3</b> Dataset</a></li>
<li class="chapter" data-level="1.4" data-path="gwas-1---data-management-and-quality-control.html"><a href="gwas-1---data-management-and-quality-control.html#genotype-data-dormat"><i class="fa fa-check"></i><b>1.4</b> Genotype Data Dormat</a>
<ul>
<li class="chapter" data-level="1.4.1" data-path="gwas-1---data-management-and-quality-control.html"><a href="gwas-1---data-management-and-quality-control.html#bim-plink-binary-variant-file"><i class="fa fa-check"></i><b>1.4.1</b> BIM: PLINK binary variant file</a></li>
<li class="chapter" data-level="1.4.2" data-path="gwas-1---data-management-and-quality-control.html"><a href="gwas-1---data-management-and-quality-control.html#fam-plink-binary-pedigree-file"><i class="fa fa-check"></i><b>1.4.2</b> FAM: PLINK binary pedigree file</a></li>
</ul></li>
<li class="chapter" data-level="1.5" data-path="gwas-1---data-management-and-quality-control.html"><a href="gwas-1---data-management-and-quality-control.html#quality-control-protocol"><i class="fa fa-check"></i><b>1.5</b> Quality control protocol</a>
<ul>
<li class="chapter" data-level="1.5.1" data-path="gwas-1---data-management-and-quality-control.html"><a href="gwas-1---data-management-and-quality-control.html#why-do-we-need-the-qc-on-gwas-data"><i class="fa fa-check"></i><b>1.5.1</b> Why do we need the QC on GWAS data?</a></li>
<li class="chapter" data-level="1.5.2" data-path="gwas-1---data-management-and-quality-control.html"><a href="gwas-1---data-management-and-quality-control.html#the-qc-protocol-from-sample-qc-to-variant-qc"><i class="fa fa-check"></i><b>1.5.2</b> The QC protocol: from Sample QC to Variant QC</a></li>
</ul></li>
<li class="chapter" data-level="1.6" data-path="gwas-1---data-management-and-quality-control.html"><a href="gwas-1---data-management-and-quality-control.html#sample-qc"><i class="fa fa-check"></i><b>1.6</b> Sample-QC</a>
<ul>
<li class="chapter" data-level="1.6.1" data-path="gwas-1---data-management-and-quality-control.html"><a href="gwas-1---data-management-and-quality-control.html#individuals-with-excessive-missing-genotypes"><i class="fa fa-check"></i><b>1.6.1</b> Individuals with excessive missing genotypes</a></li>
<li class="chapter" data-level="1.6.2" data-path="gwas-1---data-management-and-quality-control.html"><a href="gwas-1---data-management-and-quality-control.html#individuals-with-sex-discrepancy"><i class="fa fa-check"></i><b>1.6.2</b> Individuals with sex discrepancy</a></li>
<li class="chapter" data-level="1.6.3" data-path="gwas-1---data-management-and-quality-control.html"><a href="gwas-1---data-management-and-quality-control.html#individuals-with-outlying-heterozygosity-rate"><i class="fa fa-check"></i><b>1.6.3</b> Individuals with outlying heterozygosity rate</a></li>
<li class="chapter" data-level="1.6.4" data-path="gwas-1---data-management-and-quality-control.html"><a href="gwas-1---data-management-and-quality-control.html#duplicated-or-related-individuals"><i class="fa fa-check"></i><b>1.6.4</b> Duplicated or related individuals</a></li>
<li class="chapter" data-level="1.6.5" data-path="gwas-1---data-management-and-quality-control.html"><a href="gwas-1---data-management-and-quality-control.html#population-outliers"><i class="fa fa-check"></i><b>1.6.5</b> Population outliers</a></li>
</ul></li>
<li class="chapter" data-level="1.7" data-path="gwas-1---data-management-and-quality-control.html"><a href="gwas-1---data-management-and-quality-control.html#variant-qc"><i class="fa fa-check"></i><b>1.7</b> Variant-QC</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="gwas-2---association-analysis.html"><a href="gwas-2---association-analysis.html"><i class="fa fa-check"></i><b>2</b> GWAS 2 - Association analysis</a>
<ul>
<li class="chapter" data-level="2.1" data-path="gwas-2---association-analysis.html"><a href="gwas-2---association-analysis.html#objectives-1"><i class="fa fa-check"></i><b>2.1</b> Objectives</a></li>
<li class="chapter" data-level="2.2" data-path="gwas-2---association-analysis.html"><a href="gwas-2---association-analysis.html#association-analysis"><i class="fa fa-check"></i><b>2.2</b> Association Analysis</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="gwas-2---association-analysis.html"><a href="gwas-2---association-analysis.html#phenotype-data"><i class="fa fa-check"></i><b>2.2.1</b> Phenotype Data</a></li>
<li class="chapter" data-level="2.2.2" data-path="gwas-2---association-analysis.html"><a href="gwas-2---association-analysis.html#association-test-regression"><i class="fa fa-check"></i><b>2.2.2</b> Association test: regression</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="gwas-2---association-analysis.html"><a href="gwas-2---association-analysis.html#result-annotation"><i class="fa fa-check"></i><b>2.3</b> Result Annotation</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="gwas-2---association-analysis.html"><a href="gwas-2---association-analysis.html#locuszoom"><i class="fa fa-check"></i><b>2.3.1</b> LocusZoom</a></li>
<li class="chapter" data-level="2.3.2" data-path="gwas-2---association-analysis.html"><a href="gwas-2---association-analysis.html#fuma"><i class="fa fa-check"></i><b>2.3.2</b> FUMA</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Algorithms in Bioinformatics and Genomics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="gwas-2---association-analysis" class="section level1 hasAnchor" number="2">
<h1><span class="header-section-number">Chapter 2</span> GWAS 2 - Association analysis<a href="gwas-2---association-analysis.html#gwas-2---association-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="objectives-1" class="section level2 hasAnchor" number="2.1">
<h2><span class="header-section-number">2.1</span> Objectives<a href="gwas-2---association-analysis.html#objectives-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In this practical, you will learn how to perform the association analysis and the post-GWAS analysis for data interpretation.</p>
</div>
<div id="association-analysis" class="section level2 hasAnchor" number="2.2">
<h2><span class="header-section-number">2.2</span> Association Analysis<a href="gwas-2---association-analysis.html#association-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="phenotype-data" class="section level3 hasAnchor" number="2.2.1">
<h3><span class="header-section-number">2.2.1</span> Phenotype Data<a href="gwas-2---association-analysis.html#phenotype-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Besides the 6th column in PLINK .ped and .fam file, you can use another phenotype file via the command of <code>--pheno [filename]</code> in PLINK.</p>
<ul>
<li>First, let‚Äôs use R to examine the content of the <code>CAD_LDL.pheno</code> phenotype file<br></li>
</ul>
<p>üìï <strong>Q9:</strong> How many cornoary artery disease (CAD) patients and controls are there in this dataset?</p>
‚ùì <strong>Q.</strong> Is the low density lipoprotein (LDL) level normally distributed?
<details>
<summary>
Answer
</summary>
<ul>
<li>Yes. The LDL level is normally distributed and no further transformation is needed</li>
</ul>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="gwas-2---association-analysis.html#cb39-1" tabindex="-1"></a><span class="co"># ========================== R code ==========================</span></span>
<span id="cb39-2"><a href="gwas-2---association-analysis.html#cb39-2" tabindex="-1"></a><span class="co"># Plot the histogram to assess if the LDL level is normally distributed</span></span>
<span id="cb39-3"><a href="gwas-2---association-analysis.html#cb39-3" tabindex="-1"></a><span class="fu">hist</span>(pheno<span class="sc">$</span>LDL, <span class="at">freq=</span>T, <span class="at">col=</span><span class="st">&quot;darkred&quot;</span>, <span class="at">border =</span><span class="st">&quot;black&quot;</span>, <span class="at">main=</span><span class="st">&quot;Distribution of LDL&quot;</span>, <span class="at">xlab=</span><span class="st">&quot;LDL level&quot;</span>, <span class="at">ylab=</span><span class="st">&quot;Number of samples&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="gwas-2---association-analysis.html#cb40-1" tabindex="-1"></a><span class="co"># Plot the QQ plot to assess if the LDL level is normally distributed</span></span>
<span id="cb40-2"><a href="gwas-2---association-analysis.html#cb40-2" tabindex="-1"></a><span class="fu">qqnorm</span>(pheno<span class="sc">$</span>LDL)</span>
<span id="cb40-3"><a href="gwas-2---association-analysis.html#cb40-3" tabindex="-1"></a><span class="fu">qqline</span>(pheno<span class="sc">$</span>LDL)</span>
<span id="cb40-4"><a href="gwas-2---association-analysis.html#cb40-4" tabindex="-1"></a><span class="co"># ============================================================</span></span></code></pre></div>
</details>
‚ùì <strong>Q.</strong> Is there any relationship between age, LDL level and CAD?
<details>
<summary>
Answer
</summary>
<ul>
<li>CAD patients are generally older than controls (<em>P</em>=2.6x10<sup>-43</sup>) and LDL level increases with age (<em>P</em>=0.005).</li>
</ul>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="gwas-2---association-analysis.html#cb41-1" tabindex="-1"></a><span class="co"># ========================== R code ==========================</span></span>
<span id="cb41-2"><a href="gwas-2---association-analysis.html#cb41-2" tabindex="-1"></a><span class="co"># Test if age is associated with LDL and CAD</span></span>
<span id="cb41-3"><a href="gwas-2---association-analysis.html#cb41-3" tabindex="-1"></a><span class="fu">summary</span>(<span class="fu">glm</span>(LDL <span class="sc">~</span> AGE, <span class="at">data=</span>pheno))</span>
<span id="cb41-4"><a href="gwas-2---association-analysis.html#cb41-4" tabindex="-1"></a><span class="fu">summary</span>(<span class="fu">glm</span>(CAD<span class="sc">==</span><span class="dv">2</span> <span class="sc">~</span> AGE, <span class="at">family=</span><span class="st">&quot;binomial&quot;</span>, <span class="at">data=</span>pheno))</span>
<span id="cb41-5"><a href="gwas-2---association-analysis.html#cb41-5" tabindex="-1"></a><span class="fu">by</span>(pheno<span class="sc">$</span>AGE, pheno<span class="sc">$</span>CAD, summary)</span>
<span id="cb41-6"><a href="gwas-2---association-analysis.html#cb41-6" tabindex="-1"></a><span class="co"># ============================================================</span></span></code></pre></div>
</details>
</div>
<div id="association-test-regression" class="section level3 hasAnchor" number="2.2.2">
<h3><span class="header-section-number">2.2.2</span> Association test: regression<a href="gwas-2---association-analysis.html#association-test-regression" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>Next, we will use PLINK to perform association test to test for association with LDL</li>
</ul>
<div id="linear-regression-without-adjustment" class="section level4 hasAnchor" number="2.2.2.1">
<h4><span class="header-section-number">2.2.2.1</span> Linear regression WITHOUT adjustment<a href="gwas-2---association-analysis.html#linear-regression-without-adjustment" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb42"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb42-1"><a href="gwas-2---association-analysis.html#cb42-1" tabindex="-1"></a><span class="ex">~/plink</span> <span class="at">--bfile</span> chrAll.ASA.afterSampleQC.afterVariantQC <span class="dt">\</span></span>
<span id="cb42-2"><a href="gwas-2---association-analysis.html#cb42-2" tabindex="-1"></a>  <span class="at">--chr</span> 1-22,X,XY <span class="dt">\</span></span>
<span id="cb42-3"><a href="gwas-2---association-analysis.html#cb42-3" tabindex="-1"></a>  <span class="at">--pheno</span> CAD_LDL.pheno <span class="dt">\</span></span>
<span id="cb42-4"><a href="gwas-2---association-analysis.html#cb42-4" tabindex="-1"></a>  <span class="at">--pheno-name</span> LDL <span class="dt">\</span></span>
<span id="cb42-5"><a href="gwas-2---association-analysis.html#cb42-5" tabindex="-1"></a>  <span class="at">--linear</span> <span class="at">--adjust</span> <span class="dt">\</span></span>
<span id="cb42-6"><a href="gwas-2---association-analysis.html#cb42-6" tabindex="-1"></a>  <span class="at">--out</span> LDL</span></code></pre></div>
<p>By default, sex is automatically added as a covariate for all X chromosome SNPs but not anywhere else. The <code>--adjust</code> option generates an <code>.adjusted</code> file with basic multiple testing corrections (e.g.¬†FDR) for the raw p-values. It also estimates the genomic control factor (lambda) from the data.</p>
<p>üìï <strong>Q10:</strong> Which SNP is the top SNP? What is the p-value? Which allele increases LDL level? Which allele decreases LDL level? Can you give the effect size and the confidence interval (CI)?</p>
‚ùì <strong>Q.</strong> What is the lamda? Does it indicate inflation of the association statistics?
<details>
<summary>
Answer
</summary>
<ul>
<li>lamda=1.00236. No inflation in summary statistics.</li>
</ul>
</details>
<p><br>
- Plot the manhattan plot to visualize the association results</p>
<div class="sourceCode" id="cb43"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb43-1"><a href="gwas-2---association-analysis.html#cb43-1" tabindex="-1"></a><span class="co"># Rscript practical3.manhattanPlot.R &lt;assoc_file&gt; &lt;outpu_prefix&gt;</span></span>
<span id="cb43-2"><a href="gwas-2---association-analysis.html#cb43-2" tabindex="-1"></a><span class="ex">Rscript</span> practical3.manhattanPlot.R LDL.assoc.linear manhattanPlot.LDL</span></code></pre></div>
</div>
<div id="linear-regression-adjusted-for-age" class="section level4 hasAnchor" number="2.2.2.2">
<h4><span class="header-section-number">2.2.2.2</span> Linear regression adjusted for age<a href="gwas-2---association-analysis.html#linear-regression-adjusted-for-age" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>To adjust for confounding factor(s), you can use <code>--covar &lt;filename&gt;</code> to specify the covariate file, i.e.¬†<code>CAD_LDL.pheno</code> in this example. The covariate(s) used for adjustment is specified through <code>--covar-name</code>.</p>
<pre><code>~/plink --bfile chrAll.ASA.afterSampleQC.afterVariantQC \
  --chr 1-22,X,XY \
  --pheno CAD_LDL.pheno \
  --pheno-name LDL \
  <b>--covar CAD_LDL.pheno \</b>
  <b>--covar-name AGE \</b>
  <b>--hide-covar \</b>
  --linear --adjust \
  --out LDL.adj-AGE
</code></pre>
<ul>
<li>Plot the association results without adjustment of age and compare it with the previous one without adjustment</li>
</ul>
<div class="sourceCode" id="cb44"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb44-1"><a href="gwas-2---association-analysis.html#cb44-1" tabindex="-1"></a><span class="ex">Rscript</span> practical3.manhattanPlot.R LDL.adj-AGE.assoc.linear manhattanPlot.LDL.adj-AGE</span></code></pre></div>
</div>
<div id="logistic-regression-with-and-without-adjusting-for-age-to-test-for-association-with-cad" class="section level4 hasAnchor" number="2.2.2.3">
<h4><span class="header-section-number">2.2.2.3</span> Logistic regression with and without adjusting for age to test for association with CAD<a href="gwas-2---association-analysis.html#logistic-regression-with-and-without-adjusting-for-age-to-test-for-association-with-cad" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li>We can also use PLINK to perform association test to test for association with binary trait, CAD</li>
</ul>
<p>üìï <strong>Q11:</strong> Perform association test for CAD while adjusting for age. Which SNP is the top SNP? What is the p-value? What are the risk and protective alleles? Can you give the odds ratio and the confidence interval (CI)?</p>
</div>
</div>
</div>
<div id="result-annotation" class="section level2 hasAnchor" number="2.3">
<h2><span class="header-section-number">2.3</span> Result Annotation<a href="gwas-2---association-analysis.html#result-annotation" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="locuszoom" class="section level3 hasAnchor" number="2.3.1">
<h3><span class="header-section-number">2.3.1</span> LocusZoom<a href="gwas-2---association-analysis.html#locuszoom" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><a href="https://statgen.github.io/localzoom/">LocalZoom</a> (<a href="https://statgen.github.io/localzoom/" class="uri">https://statgen.github.io/localzoom/</a>) is a tool for generating regional association plot via web browser or offline. For the web-based js version without uploading the summary statistics, a tabix-indexed tab-delimited text file is needed. If you are comfortable uploading your data to a server, you can consider using the my.locuszoom.org upload service with more custom function.</p>
<p>Please noted that both the reference (Ref) and alternative (Alt) alleles are needed for visualizing the LD. For simplicity, we use A1 as the alternative allele and A2 as the reference allele here. You will usually flip to the reference strand and get the right Ref/Alt alleles after imputation.</p>
<div class="sourceCode" id="cb45"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb45-1"><a href="gwas-2---association-analysis.html#cb45-1" tabindex="-1"></a><span class="co"># (i) convert &quot;fixed width&quot; PLINK association result format to &quot;tab-delimited&quot;</span></span>
<span id="cb45-2"><a href="gwas-2---association-analysis.html#cb45-2" tabindex="-1"></a><span class="co"># (ii) add the A2 allele (specifying the right Ref and Alt alleles are needed for LD plot)</span></span>
<span id="cb45-3"><a href="gwas-2---association-analysis.html#cb45-3" tabindex="-1"></a><span class="fu">awk</span> <span class="st">&#39;BEGIN { OFS=&quot;\t&quot;; a2[&quot;SNP&quot;]=&quot;A2&quot; } NR==FNR { a2[$2]=$6 } NR!=FNR { $4=$4&quot;\t&quot;a2[$2]; if (FNR!=1) $2=$1&quot;:&quot;$3; print }&#39;</span> <span class="dt">\</span></span>
<span id="cb45-4"><a href="gwas-2---association-analysis.html#cb45-4" tabindex="-1"></a> chrAll.ASA.afterSampleQC.afterVariantQC.bim <span class="dt">\</span></span>
<span id="cb45-5"><a href="gwas-2---association-analysis.html#cb45-5" tabindex="-1"></a> LDL.adj-AGE.assoc.linear <span class="kw">|</span> <span class="dt">\</span></span>
<span id="cb45-6"><a href="gwas-2---association-analysis.html#cb45-6" tabindex="-1"></a> <span class="ex">bgzip</span> <span class="op">&gt;</span> LDL.adj-AGE.assoc.linear.forLocuszoom.gz</span>
<span id="cb45-7"><a href="gwas-2---association-analysis.html#cb45-7" tabindex="-1"></a> </span>
<span id="cb45-8"><a href="gwas-2---association-analysis.html#cb45-8" tabindex="-1"></a><span class="co"># (iii) index the association file</span></span>
<span id="cb45-9"><a href="gwas-2---association-analysis.html#cb45-9" tabindex="-1"></a><span class="ex">tabix</span> <span class="at">-s</span> 1 <span class="at">-b</span> 3 <span class="at">-e</span> 3 <span class="at">--skip-lines</span> 1 <span class="at">-f</span> LDL.adj-AGE.assoc.linear.forLocuszoom.gz</span></code></pre></div>
<ul>
<li><p>First, make sure that <code>GWAS</code> and the right genome assembly are chosen, i.e.¬†<code>GRCh37</code> for this example dataset
<img src="https://user-images.githubusercontent.com/8644480/171670740-86b20b26-cc20-4512-81cc-d43940be970c.png" width=800></p></li>
<li><p>Click <code>Add tab-indexed datafile</code> &gt; <code>Browse</code> to add the LDL.adj-AGE.assoc.linear.forLocuszoom <strong>.gz</strong> and <strong>.gz.tbi</strong> files
<img src="https://user-images.githubusercontent.com/8644480/171672531-00848874-2530-446a-aad4-102475c2ea62.png" width=800></p></li>
<li><p>Choose <code>GWAS scattered plot</code> as <code>Type</code>
<img src="https://user-images.githubusercontent.com/8644480/171672261-3cf5120d-a24b-455a-96e6-959e262b2ef6.png" width=250></p></li>
<li><p>Under <code>Variant from columns</code> &gt; Choose the major allele <code>A2</code> as <code>Ref</code> allele and the minor allele <code>A1</code> as the <code>Alt</code> allele
<img src="https://user-images.githubusercontent.com/8644480/171672887-bc1c4203-0880-44a5-9f66-64b376a155a9.png" width=500></p></li>
<li><p>Next &gt; Accept option &gt; Type <code>rs2075650</code> in the box &gt; Go to the region</p></li>
<li><p>Choose <code>LD population: EAS</code></p></li>
<li><p>You can also change the range of region to <code>19:45300000-45500000</code> for a zoom in view
<img src="https://user-images.githubusercontent.com/8644480/171678660-f12f0ff9-c4d1-4472-bf9d-089ad58d0164.png" width=800></p></li>
</ul>
</div>
<div id="fuma" class="section level3 hasAnchor" number="2.3.2">
<h3><span class="header-section-number">2.3.2</span> FUMA<a href="gwas-2---association-analysis.html#fuma" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>You can also use the SNP2GENE function in <a href="https://fuma.ctglab.nl/">FUMA</a> (Functional Mapping and Annotation of Genome-Wide Association Studies; <a href="https://fuma.ctglab.nl/" class="uri">https://fuma.ctglab.nl/</a>) for annotating, prioritizing, and interpreting the GWAS results. The SNP2GENE function takes GWAS summary statistics as input, and provides functional annotation for all SNPs in genomic areas identified by lead SNPs.</p>
<ul>
<li>SNP2GENE result for <a href="https://fuma.ctglab.nl/browse/461">CAD</a></li>
<li>SNP2GENE result for <a href="https://fuma.ctglab.nl/browse/462">LDL</a></li>
</ul>
<p>üìï <strong>Q12:</strong> For CAD, how many independent GWAS-significant loci?</p>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="gwas-1---data-management-and-quality-control.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/02-GWAS-2.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
