<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Algorithms in Bioinformatics and Genomics</title>
  <meta name="description" content="Algorithms in Bioinformatics and Genomics" />
  <meta name="generator" content="bookdown 0.42.1 and GitBook 2.6.7" />

  <meta property="og:title" content="Algorithms in Bioinformatics and Genomics" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Algorithms in Bioinformatics and Genomics" />
  
  
  

<meta name="author" content="Clara Tang" />


<meta name="date" content="2025-02-14" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  

<link rel="next" href="genomewide-association-analysis-gwas.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome</a></li>
<li class="chapter" data-level="1" data-path="genomewide-association-analysis-gwas.html"><a href="genomewide-association-analysis-gwas.html"><i class="fa fa-check"></i><b>1</b> Genomewide association analysis (GWAS)</a>
<ul>
<li class="chapter" data-level="1.1" data-path="genomewide-association-analysis-gwas.html"><a href="genomewide-association-analysis-gwas.html#objectives"><i class="fa fa-check"></i><b>1.1</b> Objectives</a></li>
<li class="chapter" data-level="1.2" data-path="genomewide-association-analysis-gwas.html"><a href="genomewide-association-analysis-gwas.html#prerequisites"><i class="fa fa-check"></i><b>1.2</b> Prerequisites</a></li>
<li class="chapter" data-level="1.3" data-path="genomewide-association-analysis-gwas.html"><a href="genomewide-association-analysis-gwas.html#step-0-download-the-genotype-data"><i class="fa fa-check"></i><b>1.3</b> Step 0: Download the genotype data</a>
<ul>
<li class="chapter" data-level="1.3.1" data-path="genomewide-association-analysis-gwas.html"><a href="genomewide-association-analysis-gwas.html#dataset"><i class="fa fa-check"></i><b>1.3.1</b> Dataset</a></li>
</ul></li>
<li class="chapter" data-level="1.4" data-path="genomewide-association-analysis-gwas.html"><a href="genomewide-association-analysis-gwas.html#genotype-data-format"><i class="fa fa-check"></i><b>1.4</b> Genotype Data format</a>
<ul>
<li class="chapter" data-level="1.4.1" data-path="genomewide-association-analysis-gwas.html"><a href="genomewide-association-analysis-gwas.html#plink-binary-variant-file-bim"><i class="fa fa-check"></i><b>1.4.1</b> - PLINK binary variant file (BIM)</a></li>
<li class="chapter" data-level="1.4.2" data-path="genomewide-association-analysis-gwas.html"><a href="genomewide-association-analysis-gwas.html#plink-binary-pedigree-file-fam"><i class="fa fa-check"></i><b>1.4.2</b> - PLINK binary pedigree file (FAM)</a></li>
</ul></li>
<li class="chapter" data-level="1.5" data-path="genomewide-association-analysis-gwas.html"><a href="genomewide-association-analysis-gwas.html#sample-qc"><i class="fa fa-check"></i><b>1.5</b> Sample-QC</a></li>
<li class="chapter" data-level="1.6" data-path="genomewide-association-analysis-gwas.html"><a href="genomewide-association-analysis-gwas.html#why-do-we-need-the-qc-on-gwas-data"><i class="fa fa-check"></i><b>1.6</b> Why do we need the QC on GWAS data?</a></li>
<li class="chapter" data-level="1.7" data-path="genomewide-association-analysis-gwas.html"><a href="genomewide-association-analysis-gwas.html#the-qc-protocol-from-sample-qc-to-variant-qc"><i class="fa fa-check"></i><b>1.7</b> The QC protocol: from Sample QC to Variant QC</a></li>
<li class="chapter" data-level="1.8" data-path="genomewide-association-analysis-gwas.html"><a href="genomewide-association-analysis-gwas.html#step-2-individuals-with-sex-discrepancy"><i class="fa fa-check"></i><b>1.8</b> Step 2: Individuals with sex discrepancy</a></li>
<li class="chapter" data-level="1.9" data-path="genomewide-association-analysis-gwas.html"><a href="genomewide-association-analysis-gwas.html#step-3-individuals-with-outlying-heterozygosity-rate"><i class="fa fa-check"></i><b>1.9</b> Step 3: Individuals with outlying heterozygosity rate</a></li>
<li class="chapter" data-level="1.10" data-path="genomewide-association-analysis-gwas.html"><a href="genomewide-association-analysis-gwas.html#step-4-duplicated-or-related-individuals"><i class="fa fa-check"></i><b>1.10</b> Step 4: Duplicated or related individuals</a></li>
<li class="chapter" data-level="1.11" data-path="genomewide-association-analysis-gwas.html"><a href="genomewide-association-analysis-gwas.html#step-5-population-outliers"><i class="fa fa-check"></i><b>1.11</b> Step 5: Population outliers</a></li>
<li class="chapter" data-level="1.12" data-path="genomewide-association-analysis-gwas.html"><a href="genomewide-association-analysis-gwas.html#variant-qc"><i class="fa fa-check"></i><b>1.12</b> Variant-QC</a></li>
<li class="chapter" data-level="1.13" data-path="genomewide-association-analysis-gwas.html"><a href="genomewide-association-analysis-gwas.html#association-analysis"><i class="fa fa-check"></i><b>1.13</b> Association analysis</a></li>
<li class="chapter" data-level="1.14" data-path="genomewide-association-analysis-gwas.html"><a href="genomewide-association-analysis-gwas.html#step-1-examining-the-phenotype-file"><i class="fa fa-check"></i><b>1.14</b> Step 1: Examining the phenotype file</a></li>
<li class="chapter" data-level="1.15" data-path="genomewide-association-analysis-gwas.html"><a href="genomewide-association-analysis-gwas.html#step-2-performing-association-analysis"><i class="fa fa-check"></i><b>1.15</b> Step 2: Performing association analysis</a>
<ul>
<li class="chapter" data-level="1.15.1" data-path="genomewide-association-analysis-gwas.html"><a href="genomewide-association-analysis-gwas.html#perform-linear-regression-without-adjusting-for-age-to-test-for-association-with-ldl"><i class="fa fa-check"></i><b>1.15.1</b> 1. Perform linear regression WITHOUT adjusting for age to test for association with LDL</a></li>
<li class="chapter" data-level="1.15.2" data-path="genomewide-association-analysis-gwas.html"><a href="genomewide-association-analysis-gwas.html#perform-linear-regression-while-adjusted-for-age"><i class="fa fa-check"></i><b>1.15.2</b> 2. Perform linear regression while adjusted for age</a></li>
<li class="chapter" data-level="1.15.3" data-path="genomewide-association-analysis-gwas.html"><a href="genomewide-association-analysis-gwas.html#perform-logistic-regression-with-and-without-adjusting-for-age-to-test-for-association-with-cad"><i class="fa fa-check"></i><b>1.15.3</b> 3. Perform logistic regression with and without adjusting for age to test for association with CAD</a></li>
<li class="chapter" data-level="1.15.4" data-path="genomewide-association-analysis-gwas.html"><a href="genomewide-association-analysis-gwas.html#step-3-annotating-the-association-findings"><i class="fa fa-check"></i><b>1.15.4</b> Step 3: Annotating the association findings</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="cross.html"><a href="cross.html"><i class="fa fa-check"></i><b>2</b> Cross-references</a>
<ul>
<li class="chapter" data-level="2.1" data-path="cross.html"><a href="cross.html#chapters-and-sub-chapters"><i class="fa fa-check"></i><b>2.1</b> Chapters and sub-chapters</a></li>
<li class="chapter" data-level="2.2" data-path="cross.html"><a href="cross.html#captioned-figures-and-tables"><i class="fa fa-check"></i><b>2.2</b> Captioned figures and tables</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Algorithms in Bioinformatics and Genomics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="header">
<h1 class="title">Algorithms in Bioinformatics and Genomics</h1>
<p class="author"><em>Clara Tang</em></p>
<p class="date"><em>2025-02-14</em></p>
</div>
<div id="welcome" class="section level1 unnumbered hasAnchor">
<h1>Welcome<a href="index.html#welcome" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Welcome to the book for practical sections of <a href="">HTI5802: Algorithms in Bioinformatics and Genomics</a>. Most contents are demonstrated with PLINK and R programming language.</p>
<p>This book is designed as a collection of R Markdown notebooks, as supplementary to the lecture notes for GWAS and sequence interpretation of HTI5802 at PolyU.</p>
<ol style="list-style-type: decimal">
<li><a href="">Genomewide association analysis (GWAS)</a> (2025-02-17)</li>
<li><a href="">Variant interpretation for clinical sequencing</a> (2025-03-10)</li>
</ol>
<p>Note: Most contents may be updated before or right after the lectures, so please refer to the updated version.</p>
<p>GitHub Repository: you can find the source files on <a href="https://github.com/claratsm/HTI5802-book">claratsm/HTI5802-book</a> and the way to re-build this book.</p>

</div>
            </section>

          </div>
        </div>
      </div>

<a href="genomewide-association-analysis-gwas.html" class="navigation navigation-next navigation-unique" aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/index.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
