<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 2 Variant interpretation for clinical sequencing | Algorithms in Bioinformatics and Genomics</title>
  <meta name="description" content="Chapter 2 Variant interpretation for clinical sequencing | Algorithms in Bioinformatics and Genomics" />
  <meta name="generator" content="bookdown 0.42.1 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 2 Variant interpretation for clinical sequencing | Algorithms in Bioinformatics and Genomics" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 2 Variant interpretation for clinical sequencing | Algorithms in Bioinformatics and Genomics" />
  
  
  

<meta name="author" content="Clara Tang" />


<meta name="date" content="2025-02-15" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="genomewide-association-analysis-gwas.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>



<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">HTI5802 Practical</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome</a></li>
<li class="chapter" data-level="1" data-path="genomewide-association-analysis-gwas.html"><a href="genomewide-association-analysis-gwas.html"><i class="fa fa-check"></i><b>1</b> Genomewide association analysis (GWAS)</a>
<ul>
<li class="chapter" data-level="1.1" data-path="genomewide-association-analysis-gwas.html"><a href="genomewide-association-analysis-gwas.html#objectives"><i class="fa fa-check"></i><b>1.1</b> Objectives</a></li>
<li class="chapter" data-level="1.2" data-path="genomewide-association-analysis-gwas.html"><a href="genomewide-association-analysis-gwas.html#prerequisites"><i class="fa fa-check"></i><b>1.2</b> Prerequisites</a></li>
<li class="chapter" data-level="1.3" data-path="genomewide-association-analysis-gwas.html"><a href="genomewide-association-analysis-gwas.html#dataset"><i class="fa fa-check"></i><b>1.3</b> Dataset</a></li>
<li class="chapter" data-level="1.4" data-path="genomewide-association-analysis-gwas.html"><a href="genomewide-association-analysis-gwas.html#genotype-data-dormat"><i class="fa fa-check"></i><b>1.4</b> Genotype Data Dormat</a>
<ul>
<li class="chapter" data-level="1.4.1" data-path="genomewide-association-analysis-gwas.html"><a href="genomewide-association-analysis-gwas.html#bim-plink-binary-variant-file"><i class="fa fa-check"></i><b>1.4.1</b> BIM: PLINK binary variant file</a></li>
<li class="chapter" data-level="1.4.2" data-path="genomewide-association-analysis-gwas.html"><a href="genomewide-association-analysis-gwas.html#fam-plink-binary-pedigree-file"><i class="fa fa-check"></i><b>1.4.2</b> FAM: PLINK binary pedigree file</a></li>
</ul></li>
<li class="chapter" data-level="1.5" data-path="genomewide-association-analysis-gwas.html"><a href="genomewide-association-analysis-gwas.html#quality-control-protocol"><i class="fa fa-check"></i><b>1.5</b> Quality control protocol</a>
<ul>
<li class="chapter" data-level="1.5.1" data-path="genomewide-association-analysis-gwas.html"><a href="genomewide-association-analysis-gwas.html#why-do-we-need-the-qc-on-gwas-data"><i class="fa fa-check"></i><b>1.5.1</b> Why do we need the QC on GWAS data?</a></li>
<li class="chapter" data-level="1.5.2" data-path="genomewide-association-analysis-gwas.html"><a href="genomewide-association-analysis-gwas.html#the-qc-protocol-from-sample-qc-to-variant-qc"><i class="fa fa-check"></i><b>1.5.2</b> The QC protocol: from Sample QC to Variant QC</a></li>
</ul></li>
<li class="chapter" data-level="1.6" data-path="genomewide-association-analysis-gwas.html"><a href="genomewide-association-analysis-gwas.html#sample-qc"><i class="fa fa-check"></i><b>1.6</b> Sample-QC</a>
<ul>
<li class="chapter" data-level="1.6.1" data-path="genomewide-association-analysis-gwas.html"><a href="genomewide-association-analysis-gwas.html#individuals-with-excessive-missing-genotypes"><i class="fa fa-check"></i><b>1.6.1</b> Individuals with excessive missing genotypes</a></li>
<li class="chapter" data-level="1.6.2" data-path="genomewide-association-analysis-gwas.html"><a href="genomewide-association-analysis-gwas.html#individuals-with-sex-discrepancy"><i class="fa fa-check"></i><b>1.6.2</b> Individuals with sex discrepancy</a></li>
<li class="chapter" data-level="1.6.3" data-path="genomewide-association-analysis-gwas.html"><a href="genomewide-association-analysis-gwas.html#individuals-with-outlying-heterozygosity-rate"><i class="fa fa-check"></i><b>1.6.3</b> Individuals with outlying heterozygosity rate</a></li>
<li class="chapter" data-level="1.6.4" data-path="genomewide-association-analysis-gwas.html"><a href="genomewide-association-analysis-gwas.html#duplicated-or-related-individuals"><i class="fa fa-check"></i><b>1.6.4</b> Duplicated or related individuals</a></li>
<li class="chapter" data-level="1.6.5" data-path="genomewide-association-analysis-gwas.html"><a href="genomewide-association-analysis-gwas.html#population-outliers"><i class="fa fa-check"></i><b>1.6.5</b> Population outliers</a></li>
</ul></li>
<li class="chapter" data-level="1.7" data-path="genomewide-association-analysis-gwas.html"><a href="genomewide-association-analysis-gwas.html#variant-qc"><i class="fa fa-check"></i><b>1.7</b> Variant-QC</a></li>
<li class="chapter" data-level="1.8" data-path="genomewide-association-analysis-gwas.html"><a href="genomewide-association-analysis-gwas.html#association-analysis"><i class="fa fa-check"></i><b>1.8</b> Association Analysis</a>
<ul>
<li class="chapter" data-level="1.8.1" data-path="genomewide-association-analysis-gwas.html"><a href="genomewide-association-analysis-gwas.html#phenotype-data"><i class="fa fa-check"></i><b>1.8.1</b> Phenotype Data</a></li>
<li class="chapter" data-level="1.8.2" data-path="genomewide-association-analysis-gwas.html"><a href="genomewide-association-analysis-gwas.html#association-test-regression"><i class="fa fa-check"></i><b>1.8.2</b> Association test: regression</a></li>
</ul></li>
<li class="chapter" data-level="1.9" data-path="genomewide-association-analysis-gwas.html"><a href="genomewide-association-analysis-gwas.html#result-annotation"><i class="fa fa-check"></i><b>1.9</b> Result Annotation</a>
<ul>
<li class="chapter" data-level="1.9.1" data-path="genomewide-association-analysis-gwas.html"><a href="genomewide-association-analysis-gwas.html#locuszoom"><i class="fa fa-check"></i><b>1.9.1</b> LocusZoom</a></li>
<li class="chapter" data-level="1.9.2" data-path="genomewide-association-analysis-gwas.html"><a href="genomewide-association-analysis-gwas.html#fuma"><i class="fa fa-check"></i><b>1.9.2</b> FUMA</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="variant-interpretation-for-clinical-sequencing.html"><a href="variant-interpretation-for-clinical-sequencing.html"><i class="fa fa-check"></i><b>2</b> Variant interpretation for clinical sequencing</a>
<ul>
<li class="chapter" data-level="2.1" data-path="variant-interpretation-for-clinical-sequencing.html"><a href="variant-interpretation-for-clinical-sequencing.html#objectives-1"><i class="fa fa-check"></i><b>2.1</b> Objectives</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Algorithms in Bioinformatics and Genomics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="variant-interpretation-for-clinical-sequencing" class="section level1 hasAnchor" number="2">
<h1><span class="header-section-number">Chapter 2</span> Variant interpretation for clinical sequencing<a href="variant-interpretation-for-clinical-sequencing.html#variant-interpretation-for-clinical-sequencing" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="objectives-1" class="section level2 hasAnchor" number="2.1">
<h2><span class="header-section-number">2.1</span> Objectives<a href="variant-interpretation-for-clinical-sequencing.html#objectives-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>To be updated</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="genomewide-association-analysis-gwas.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/02-VariantInterpretation.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
